packages/core/src/utils.ts

```diff
@@ -479,3 +483,179 @@ export function capitalize(word: string): string {
 export function isLowerCase(c: string): boolean {
   return c === c.toLowerCase();
 }
+
+/**
+ * Tries to find a code string for a given system within a given codeable concept.
+ * @param concept The codeable concept.
+ * @param system The system string.
+ * @returns The code if found; otherwise undefined.
+ */
+export function getCodeBySystem(concept: CodeableConcept, system: string): string | undefined {
+  return concept?.coding?.find((coding) => coding.system === system)?.code;
+}
+
+/**
+ * Sets a code for a given system within a given codeable concept.
+ * @param concept The codeable concept.
+ * @param system The system string.
+ * @param code The code value.
+ */
+export function setCodeBySystem(concept: CodeableConcept, system: string, code: string): void {
+  if (!concept.coding) {
+    concept.coding = [];
+  }
+  const coding = concept.coding?.find((coding) => coding.system === system);
+  if (coding) {
+    coding.code = code;
+  } else {
+    concept.coding?.push({ system, code });
+  }
+}
+
+/**
+ * Tries to find an observation interval for the given patient and value.
+ * @param definition The observation definition.
+ * @param patient The patient.
+ * @param value The observation value.
+ * @returns The observation interval if found; otherwise undefined.
+ */
+export function findObservationInterval(
```

Comment:
This should probably return an array, in case a value falls inside multiple intervals. This could happen if the OD defines a measureable range as well as a normal range

Reply:
We could also potentially accept an object with props with things to filter the return results by. For example: 

`findObservationInterval(definition, patient, value, {category})`

Or we could say that multiple results only happen for ranges in different categories, and we can have an optional `category` param that defaults to `reference`


https://www.hl7.org/fhir/valueset-observation-range-category.html